src/main/java/bio/terra/workspace/common/exception/ControlledResourceNotFoundException.java

```diff
@@ -0,0 +1,26 @@
+package bio.terra.workspace.common.exception;
```

Comment:
I'm not sure we need to separate `ControlledResourceNotFoundException` from `DataReferenceNotFoundException`. Thoughts on just renaming the existing exception to `ResourceNotFoundException` and using that?

Reply:
That works, if we're settled on resource as the top-level name.